src/telemetryReporter.ts

```diff
@@ -0,0 +1,12 @@
+"use strict";
+
+import TelemetryReporter from 'vscode-extension-telemetry';
+import { Constants } from './constants';
+import { ExtensionContext } from 'vscode';
+
+export function createReporter(context: ExtensionContext) {
```

Comment:
wouldn't it be possible just to create global reporter object here, and export that global instance?
This way it wouldn't be necessary to pass it as a parameter.